bin/tx-decoder/src/queries.rs

```diff
@@ -32,13 +32,54 @@ pub struct BlockModel {
 	pub ext: Vec<u8>,
 	pub spec: i32,
 }
-// just returns all blocks in the database of a specific spec as a stream
-pub fn blocks<'a>(conn: &'a mut PgConnection, spec: i32) -> impl Stream<Item = Result<BlockModel, Error>> + 'a {
+
+#[derive(FromRow)]
+struct Count {
+	count: i64
+}
+
+/// returns how many blocks exist for a spec version.
+pub async fn blocks_in_spec(conn: &mut PgConnection, spec: i32) -> Result<i64, Error> {
+	Ok(sqlx::query_as::<_, Count>("SELECT COUNT(*) FROM blocks WHERE spec = $1")
+		.bind(spec)
+		.fetch_one(conn)
+		.await?
+		.count
+	)
+}
+
+pub async fn total_block_count(conn: &mut PgConnection) -> Result<i64, Error> {
+	Ok(sqlx::query_as::<_, Count>("SELECT COUNT(*) FROM blocks")
+		.fetch_one(conn)
+		.await?
+		.count
+	)
+}
+
```

Comment:
```suggestion
```suggestion

/// returns how many blocks exist up to a spec version
```
```